### ISSUE TITLE
Review conda setup 

### ISSUE THREAD

**Issue by OwenArnold** [scipp/scipp#2085]
We are not using much from V3 https://bioconda.github.io/contributor/cb3.html

Things to consider (not full list)

* Host sections
* Native compiler support (OSX, Windows), though last check did not look good
* Better ways of pinning
* strong/weak runtime exports
* Better way to use ccache (get back to fully integrated `conda build` approach to building)

Note that our CI is split between the package common repo https://github.com/scipp/pipelines/ and the contents of `conda` and `tools` in the respective package repos.

**Comment by Pasarus** [comment]
All these changes can be done in the meta.yaml file, for each individual project.

"The new build section should have things like compilers, git, automake, make, cmake, and other build tools." E.g.:
```
- {{ compiler("c") }}
- {{ compiler("cxx") }}
- ninja  # [unix]
- cmake
- python
```

"The new host section should have everything else." e.g.:
```
- tbb-devel
```

"The run section remains the same." e.g.:
```
- appdirs
- numpy >=1.15.3
- python {{ python }}
- python-configuration
- pyyaml
- tbb
```

Native compiler support is available for OSX via LLVM and Linux via GCC from Conda which should be sufficient for conda build. The entry to build  `{{ compiler('cxx') }}` will sort getting the correct C++ compiler for Linux and OSX for us. I do not believe this will ever be possible with Windows so we still need to install Visual Studio separately, as long as we are using MSVC anyway.

Not sure what you mean by `better ways of pinning` my guess is handling build and runtime dependency pinning, I think conda build 3 does this well for us anyway according to the linked contributors documentation all we have to do is make sure that the package is in the host and run sections and it will pin the versions automatically, not sure how it handles `tbb-dev` and `tbb` but I assume it handles it just fine as `tbb` is a dependency for `tbb-dev`.
Alternatively we can "For example, you may use Pinning expressions to obtain flexible version pinning relative to versions present at build time:" such as
```
- {{ pin_subpackage('libpng', max_pin='x.x') }}
```

In order to handle weak runtime requirements, we can either just not list them, or if we need to specify a specific version we can add a section called: `run_constrained` inside of `requirements` which will allow version pinning of these exports if they are going to be present. So for example we could force Mantid Framework to be versions `<6`:
```
requirements:
  run_constrained:
    - mantid-framework <6
```

Pybind 11 has some use cases where the ABIs have to match between packages to function properly so it's recommend to include:
```
requirements:
  host:
    - pybind11-abi
```
I don't see it presently listed as a used package for either runtime or build/host at the moment, so I am a little confused, as to how we properly handle this in Conda build.

**Comment by SimonHeybrock** [comment]
> I don't see it [pybind11] presently listed as a used package for either runtime or build/host at the moment, so I am a little confused, as to how we properly handle this in Conda build.

We are installing it using conan since a couple weeks ago, conda is not involved there any more.

**Comment by OwenArnold** [comment]
Moving OSX fully over to clang would be a good thing to try. Though Last time I looked it involved some hacky way of getting the XCode SDK.

**Comment by jl-wynen** [comment]
conda build 3.0 features and cross compilation support were implemented in #2550

conda-forge global pinning is currently not possible because conda-forge pins numpy to 1.19 but we need 1.20 for tpying. See #2559
